feconf.py

```diff
@@ -235,6 +237,11 @@ def get_empty_ratings():
     },
 }
 
+# Prefixes for activity ids to denote the type of activity. These values should
+# not be changed.
+ACTIVITY_ID_PREFIX_EXPLORATION = 'e:'
```

Comment:
I thought about this more, and I agree with you. I'll refactor accordingly.

I am a bit on the fence about having a domain object for ActivityType -- it might not be necessary. I'll do the refactor and see how it goes. However, I do think that, at some point, we should have a domain object for ActivitySummary, since this is starting to emerge as a general concept, and we do much more stuff with "activity summaries" than exploration/collection summaries in isolation.

That said, thanks for pushing back on this. I think, in the back of my mind, I was concerned about query-ability -- JSON stuff isn't really queryable, but strings are, so you could do a query for is_activity_featured(activity_id). Having said that, on further reflection, I don't see why we would query -- it's probably better to just denormalize things and add a mapping that goes the other way if that becomes a use case for which we need efficiency.

So, overall, I agree with you, and I'll refactor and resubmit. Thanks!